src/lib/operators/join_hash.cpp

```diff
@@ -337,15 +347,16 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
           _probe_input_table, _column_ids.second, histograms_probe_column, _radix_bits, probe_side_bloom_filter,
           build_side_bloom_filter);
     }
+    _performance.set_step_runtime(OperatorSteps::Materialization, timer_materialization.lap());
```

Comment:
I wonder if we want to add a `timer.finish()` method, which would not allow any more calls to lap or finish. Otherwise, `timer_materialization` may falsely be reused. It also gives us some limited thread-safety.

Reply:
I see your point but I am  not sure if isn't a bit too much overhead. It's not a lot added, but I guess the idea of the Timers is to allow timing with the absolute minimum overhead possible.